pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -2795,6 +2802,111 @@ protected void internalSetNamespaceResourceGroup(String rgName) {
+                userMetadata.putAll(md.getUserMetadata());
+                String clusterName = userMetadata.get(LedgerOffloader.METADATA_PULSAR_CLUSTER_NAME);
+                if (localClusterName.equals(clusterName)) {
+                    String managedLedgerName = userMetadata.get("managedledgername");
+                    if (managedLedgerName != null) {
+                        objectInfo.put("managedLedgerName", managedLedgerName);
+                        try {
+                            status = checkLedgerShouldBeOnTieredStorage(md.getLedgerId(), md.getUuid(),
+                                    managedLedgerName, objectInfo, pulsar().getManagedLedgerFactory());
+                        } catch (InterruptedException err) {
+                            Thread.currentThread().interrupt();
+                            throw new RuntimeException(err);
+                        } catch (ManagedLedgerException err) {
+                            log.error("Error while checking managed ledger {}", managedLedgerName);
+                            throw new RuntimeException(err);
+                        }
+                    }
+                }
+            }
+            totalCount.incrementAndGet();
+            objectInfo.put("status", status);
+            switch (status) {
+                case "OK":
+                    break;
+                case "UNKNOWN":
+                    totalUnknown.incrementAndGet();
+                    break;
+                default:
+                    totalErrors.incrementAndGet();
+                    break;
+            }
+
+            objects.add(objectInfo);
+            return true;
+        }), managedLedgerOffloader.getOffloadDriverMetadata());
+        topLevelResult.put("errors", totalErrors.intValue());
+        topLevelResult.put("total", totalCount.intValue());
+        topLevelResult.put("unknownObjects", totalUnknown.intValue());
+        log.info("internalScanOffloadedLedgers {} scan finished");
+
+        return topLevelResult;
+    }
+
+    private static String checkLedgerShouldBeOnTieredStorage(long ledgerId, String uuid,
+                                                      String managedLedgerName,
+                                                      Map<String, Object> data,
+                                                      ManagedLedgerFactory managedLedgerFactory)
+            throws InterruptedException, ManagedLedgerException  {
+        try {
+            ManagedLedgerInfo managedLedgerInfo = managedLedgerFactory.getManagedLedgerInfo(managedLedgerName);
+            ManagedLedgerInfo.LedgerInfo ledgerInfo = managedLedgerInfo
+                    .ledgers.stream().filter(l -> l.ledgerId == ledgerId).findAny().orElse(null);
+            if (ledgerInfo == null) {
+                log.info("Managed ledger {} does not contain ledger {}", managedLedgerName, ledgerId);
+                return "NOT-FOUND";
```

Comment:
Why not use constant or Enum?

Reply:
I moved this code to a separate class and created constants @zymap